src/AhRulesBot/AhRulesBot/BotRequestsProcessing/BotRequestsWorker.cs

```diff
@@ -42,12 +42,12 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 
                 await foreach (Update update in updateReceiver.YieldUpdatesAsync())
                 {
-                    await _telegramMessageProcessor.Process(update.Message);
+                    await _telegramMessageProcessor.Process(update.Message, stoppingToken);
                 }
             }
             catch (Exception e)
             {
-                _logger.Error(e, "{nameof(Worker)}.{nameof(ExecuteAsync)} Some unhandled error occurred, stopping application.");
+                _logger.Error(e, "{Class}.{Method} Some unhandled error occurred, stopping application.", nameof(BotRequestsWorker), nameof(ExecuteAsync));
```

Comment:
may for for now, but I would suggest to use some enrichers to do it automatically

Reply:
Added to future improvements thanks.